src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -377,20 +377,22 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		if (!this._cursorOnNextLine() || !this._cursorLineLooksLikeWindowsPrompt()) {
 			this._windowsPromptPollingInProcess = true;
 			// Poll for 200ms until the cursor position is correct.
-			let i = 0;
-			for (; i < 20; i++) {
-				await timeout(10);
-				if (!this._windowsPromptPollingInProcess || this._cursorOnNextLine() && this._cursorLineLooksLikeWindowsPrompt()) {
-					if (!this._windowsPromptPollingInProcess) {
-						this._logService.debug('CommandDetectionCapability#_handleCommandStartWindows polling cancelled');
+			this._register(toDisposable(async () => {
```

Comment:
This would start the polling when `CommandDetectionCapability` gets disposed

Reply:
🤦🏻‍♀️